src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnectorConfig.java

```diff
@@ -0,0 +1,275 @@
+    try {
+      Method method = row.getClass().getMethod(createGetterName(fieldName));
+      return method.invoke(row);
+    } catch (Exception e) {
+      LOG.error(String.format("Error while fetching %s.%s value", row.getClass().getSimpleName(), fieldName), e);
+      return null;
+    }
+  }
+  private String createGetterName(String name) {
+    StringBuilder sb = new StringBuilder("get");
+    sb.append(name.substring(0, 1).toUpperCase());
+    sb.append(name.substring(1));
+    return sb.toString();
+  }
+  @VisibleForTesting
+  String convertToStringValue(Object fieldValue) {
+    return String.valueOf(fieldValue);
+  }
+
+  @VisibleForTesting
+  Double convertToDoubleValue(Object fieldValue) {
+    if (fieldValue instanceof String && Strings.isNullOrEmpty(String.valueOf(fieldValue))) {
+      return null;
+    }
+
+    return Double.parseDouble(String.valueOf(fieldValue));
+  }
+
+  @VisibleForTesting
+  Integer convertToIntegerValue(Object fieldValue) {
+    if (fieldValue instanceof String && Strings.isNullOrEmpty(String.valueOf(fieldValue))) {
+      return null;
+    }
+
+    return Integer.parseInt(String.valueOf(fieldValue));
+  }
+
+  @VisibleForTesting
+  Boolean convertToBooleanValue(Object fieldValue) {
+    if (fieldValue instanceof String && Strings.isNullOrEmpty(String.valueOf(fieldValue))) {
+      return null;
+    }
+
+    return Boolean.parseBoolean(String.valueOf(fieldValue));
+  }
+  @VisibleForTesting
+  Object convertToValue(String fieldName, Schema fieldSchema, Object fieldValue) {
+    Schema.Type fieldType = fieldSchema.getType();
+    Schema.LogicalType logicalType = fieldSchema.getLogicalType();
+    if (fieldSchema.getLogicalType() != null) {
+      return transformLogicalType(fieldName, logicalType, fieldValue);
+    }
+
+    switch (fieldType) {
+      case STRING:
+        return convertToStringValue(fieldValue);
+      case DOUBLE:
+        return convertToDoubleValue(fieldValue);
+      case INT:
+        return convertToIntegerValue(fieldValue);
+      case BOOLEAN:
+        return convertToBooleanValue(fieldValue);
+      case UNION:
+        if (fieldSchema.isNullable()) {
+          return convertToValue(fieldName, fieldSchema.getNonNullable(), fieldValue);
+        }
+        throw new IllegalStateException(
+                String.format("Field '%s' is of unexpected type '%s'. Declared 'complex UNION' types: %s",
+                        fieldName, fieldValue.getClass().getSimpleName(), fieldSchema.getUnionSchemas()));
+      default:
+        throw new IllegalStateException(
+                String.format("Record type '%s' is not supported for field '%s'", fieldType.name(), fieldName));
+    }
+  }
+  public void convertRecord(SourceObject sourceObject, StructuredRecord.Builder recordBuilder, ETApiObject row)
+          throws ETSdkException {
+    MarketingCloudClient client =  MarketingCloudClient.create(getClientId(), getClientSecret(), getAuthEndpoint(),
```

Comment:
For every record it is making creating MarketingCloudClient which is not a good practice, remove this line and get this client from calling method itself

Reply:
Got it. 
Done the changes as per the requirement